File: runtime/peerbot/src/com/newbound/p2p/protocol/NORELAY.java
package com.newbound.p2p.protocol;

import org.json.JSONObject;

import com.newbound.p2p.P2PService;
import com.newbound.robot.Callback;

public class NORELAY implements Callback {

	P2PService P2P;

	public NORELAY(P2PService p2p) 
	{
		super();
		P2P = p2p;
	}

	@Override
	public void execute(JSONObject data) 
	{
		try
		{
			String relay = data.getString("uuid");
			byte[] ba = (byte[])data.get("data");

			String target = new String(ba, 0, 36);
			System.out.println("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
			System.out.println(" REMOVING RELAY "+relay+" from "+target);
			System.out.println("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
			try { P2P.removeRelay(target, relay); } catch (Exception x) { x.printStackTrace(); }
		}
		catch (Exception x) { x.printStackTrace(); }
	}

}

File: runtime/peerbot/src/newbound/p2p/protocol/norelay.py
class NORELAY(object):
    def __init__(self, service):
        self.service = service

    def execute(self, data):
        relay = data['uuid']
        target = data['data'][:36].decode()
        sock = data['p2psocket']
        print('!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!')
        print(' REMOVING RELAY '+relay+' from '+target)
        print('!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!')
        sock.service.p2p.removeRelay(target, relay)
